src/Surface.cpp

```diff
@@ -901,11 +902,12 @@ int ZCylinder::intersection(Point* point, double angle, Point* points) {
      * rearrange to put in the form of the quadratic formula:
      * ax^2 + bx + c = 0
      */
-    double m = sin(angle) / cos(angle);
+    bool right = angle < M_PI / 2. || angle > 3. * M_PI / 2.;
+    double m = tan(angle);
     q = y0 - m * x0;
-    a = _A + _B * _B * m * m;
-    b = 2 * _B * m * q + _C + _D * m;
-    c = _B * q * q + _D * q + _E;
+    a = 1. + m * m;
+    b = 2 * m * q + _C + _D * m;
```

Comment:
looks weird to multiply by `1.` and `2` in successive lines - either remove the decimal from the first line or add it to the second line (understandably it doesn't matter in C/C++, but for the sake of consistency).